luthier/src/test/groovy/com/yahoo/bard/webservice/config/luthier/ResourceNodeSupplierTest.groovy

```diff
@@ -0,0 +1,92 @@
+package com.yahoo.bard.webservice.config.luthier
+
+import com.fasterxml.jackson.databind.node.ArrayNode
+import com.fasterxml.jackson.databind.node.ObjectNode
+import spock.lang.Specification
+
+class ResourceNodeSupplierTest extends Specification {
+    ResourceNodeSupplier testResourceNodeSupplier
+
+    void setup() {
+        testResourceNodeSupplier = new ResourceNodeSupplier("DimensionConfig.json")
+    }
+
+    def "default ResourceNodeSupplier only returns null when supplied null"() {
+        given:
+            ResourceNodeSupplier defaultResourceNodeSupplier = new ResourceNodeSupplier(null)
+        when:
+            ObjectNode defaultNode = defaultResourceNodeSupplier.get()
+            ObjectNode dimensionsNode = testResourceNodeSupplier.get()
+        then:
+            defaultNode == null
+            dimensionsNode != null
+    }
+
+    def "a LuthierFactoryException is thrown when .json is not formatted correctly"() {
+        given:
+            ResourceNodeSupplier nonExistentResourceNodeSupplier = new ResourceNodeSupplier("NON_EXISTENT.json")
+        when:
+            ObjectNode nonExistentNode = nonExistentResourceNodeSupplier.get()
+        then:
+            thrown(LuthierFactoryException)
+    }
+
+    def "All contents of a test dimension is correct"() {
+        when:
+            ObjectNode node = testResourceNodeSupplier.get().get("testDimension")
+            String apiName = node.get("apiName").textValue()
+            String longName = node.get("longName").textValue()
+            ArrayNode fields = node.get("fields")
+            String category = node.get("category").textValue()
+            String description = node.get("description").textValue()
+            String searchProvider = node.get("searchProvider").textValue()
+            String keyValueStore = node.get("keyValueStore").textValue()
+        then:
+            apiName == "testDimension"
+            longName == "a longName for testing"
+            fields.size() == 4
+            fields.get(0).get("name").textValue() == "TEST_PK"
```

Comment:
Might be a bit better to use a for loop for this:
```groovy
List expectedFieldNames = ["primaryKey", "TEST_FIELD_1", "TEST_FIELD_2", "TEST_FIELD_3"]
for (int i = 0; i < fields.size(); i++) {
fielsds.get(i).get("name").textValue() == expectedFieldNames[i]

```
A bit more concise and scales better if we decide to add additional fields or whatever.

Reply:
Addressed.

Reply:
Done.